### PR TITLE
Make Base files and PermissionFilter private

### PR DESCRIPTION
**Pull request by renonick87**
This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Summary
Base files and PermissionFilter are made private.

### CODE REVIEW

File: lib/js/src/manager/permission/_PermissionFilter.js [hunk 1]
```diff
@@ -35,10 +35,11 @@ import { uuid } from '../../util/uuid.js';
 /**
```
Comment (crokita) [inline lib/js/src/manager/permission/_PermissionFilter.js:35]: Missed spots for changing to _PermissionFilter for the set methods:

`@returns {PermissionFilter}`

Comment (renonick87) [reply lib/js/src/manager/permission/_PermissionFilter.js:35]: docs are now updated

File: lib/js/src/manager/permission/_PermissionManagerBase.js [hunk 2]
```diff
@@ -330,15 +331,15 @@ class PermissionManagerBase extends SubManagerBase {
     }
 
     /**
-     * Call the listener of a specific PermissionFilter
+     * Call the listener of a specific_PermissionFilter
```
Comment (crokita) [inline lib/js/src/manager/permission/_PermissionManagerBase.js:334]: needs space between specific and PermissionFilter

File: lib/js/src/manager/permission/_PermissionManagerBase.js [hunk 3]
```diff
@@ -330,15 +331,15 @@ class PermissionManagerBase extends SubManagerBase {
...
      * @private
-     * @param {PermissionFilter} permissionFilter - The PermissionFilter to invoke
-     * @returns {PermissionManagerBase} - A reference to this instance to support method chaining.
+     * @param {_PermissionFilter} permissionFilter - The_PermissionFilter to invoke
```
Comment (crokita) [inline lib/js/src/manager/permission/_PermissionManagerBase.js:336]: needs space between The and PermissionFilter

File: lib/js/src/manager/permission/_PermissionManagerBase.js [hunk 4]
```diff
@@ -350,15 +351,15 @@ class PermissionManagerBase extends SubManagerBase {
      * @returns {String} - A UUID for the Permission Filter listener. It can be used to remove the listener later.
      */
     addListener (permissionElements, groupType, listener) {
-        const filter = new PermissionFilter(null, permissionElements, groupType, listener);
+        const filter = new _PermissionFilter(null, permissionElements, groupType, listener);
         this._filters.push(filter);
         return filter.getId();
```
Comment (crokita) [inline lib/js/src/manager/permission/_PermissionManagerBase.js:356]: This should use `_getId`

File: lib/js/src/manager/permission/_PermissionManagerBase.js [hunk 5]
```diff
@@ -350,15 +351,15 @@ class PermissionManagerBase extends SubManagerBase {
...
     }
 
     /**
      * Removes specific listener
      * @param {String} filterUuid - The UUID of the listener to be removed
-     * @returns {PermissionManagerBase} - A reference to this instance to support method chaining.
+     * @returns {_PermissionManagerBase} - A reference to this instance to support method chaining.
      */
     removeListener (filterUuid) {
```
Comment (crokita) [inline lib/js/src/manager/permission/_PermissionManagerBase.js:364]: This should use `_getId` in this function

File: lib/js/src/manager/permission/_PermissionManagerBase.js [hunk 6]
```diff
@@ -124,7 +125,7 @@ class PermissionManagerBase extends SubManagerBase {
      * @param {HMILevel} previousHmiLevel - The previous HMI Level enum value.
      * @param {PermissionItem[]} currentPermissionItems - An array of current PermissionItem instances.
      * @param {HMILevel} currentHmiLevel - The current HMI Level enum value.
-     * @returns {PermissionManagerBase} - A reference to this instance to support method chaining.
+     * @returns {_PermissionManagerBase} - A reference to this instance to support method chaining.
      */
     _notifyListeners (previousPermissionItems, previousHmiLevel, currentPermissionItems, currentHmiLevel) {
         for (const filter of this._filters) {
```
Comment (crokita) [inline lib/js/src/manager/permission/_PermissionManagerBase.js:131]: getPermissionElements to _getPermissionElements within this function. same with getGroupType

**Pull request merged**